presto-main/src/main/java/com/facebook/presto/split/MappedPageSource.java

```diff
@@ -63,6 +63,9 @@ public boolean isFinished()
     @Override
     public Page getNextPage()
     {
+        if (isFinished()) {
```

Comment:
Technically, delegate.getNextPage() could return null even if isFinished is false, so it'd be better to just check whether `nextPage` below is null.